docs/add-ins/get-or-set-the-location-of-an-appointment.md

```diff
@@ -1,30 +1,111 @@
 
 | API | Applicable appointment modes |
 |---|---|
-| [item.location](/javascript/api/outlook/office.appointmentread#location) | Attendee/Read |
-| [item.location.getAsync](/javascript/api/outlook/office.location#getasync-options--callback-) | Organizer/Compose |
-| [item.location.setAsync](/javascript/api/outlook/office.location#setasync-location--options--callback-) | Organizer/Compose |
-| [item.enhancedLocation.getAsync](/javascript/api/outlook/office.enhancedlocation#getasync-options--callback-) | Organizer/Compose,<br>Attendee/Read |
-| [item.enhancedLocation.addAsync](/javascript/api/outlook/office.enhancedlocation#addasync-locationidentifiers--options--callback-) | Organizer/Compose |
-| [item.enhancedLocation.removeAsync](/javascript/api/outlook/office.enhancedlocation#removeasync-locationidentifiers--options--callback-) | Organizer/Compose |
+| [item.location](/javascript/api/outlook/office.appointmentread?view=outlook-js-1.1#location) | Attendee/Read |
+| [item.location.getAsync](/javascript/api/outlook/office.location?view=outlook-js-1.1#getasync-options--callback-) | Organizer/Compose |
+| [item.location.setAsync](/javascript/api/outlook/office.location?view=outlook-js-1.1#setasync-location--options--callback-) | Organizer/Compose |
+| [item.enhancedLocation.getAsync](/javascript/api/outlook/office.enhancedlocation?view=outlook-js-1.8#getasync-options--callback-) | Organizer/Compose,<br>Attendee/Read |
+| [item.enhancedLocation.addAsync](/javascript/api/outlook/office.enhancedlocation?view=outlook-js-1.8#addasync-locationidentifiers--options--callback-) | Organizer/Compose |
+| [item.enhancedLocation.removeAsync](/javascript/api/outlook/office.enhancedlocation?view=outlook-js-1.8#removeasync-locationidentifiers--options--callback-) | Organizer/Compose |
 
 To use the methods that are available only to compose add-ins, configure the add-in manifest to activate the add-in in Organizer/Compose mode. See [Create Outlook add-ins for compose forms](compose-scenario.md) for more details.
 
+## Use the `enhancedLocation` API
+
+You can use the `enhancedLocation` API to get and set an appointment's location. The location field supports multiple locations and, for each location, you can set the display name, type, and email address (only applicable for conference rooms). At present, supported [location types](/javascript/api/outlook/office.mailboxenums.locationtype) are `Room` (i.e., a valid conference room or similar resource) and `Custom` (represents any other type of location).
+
+### Add location
+
+The following example shows how to add a location by calling [addAsync](/javascript/api/outlook/office.enhancedlocation#addasync-locationidentifiers--options--callback-) on [mailbox.item.enhancedLocation](/javascript/api/outlook/office.appointmentcompose#enhancedlocation).
+
+```js
+var item;
+var locations = [
+    {
+        "id": "Contoso",
+        "type": Office.MailboxEnums.LocationType.Custom
```

Comment:
Would it be more helpful to users to set the location as a room and set all the room-related parameters?

Reply:
possibly - however, for the room option to work, they need a legit email address for their situation which I can't guarantee